File: languages/java/src/main/java/com/bloock/sdk/entity/identity/DidMethod.java
package com.bloock.sdk.entity.identity;

import com.bloock.sdk.bridge.proto.BloockIdentityEntities;

/**
 * Represents the type of method did.
 */
public enum DidMethod {
  /**
   * Represents the polygon id method did.
   */
  PolygonID,
  /**
   * Represents the polygon id test method did.
   */
  PolygonIDTest;

  public static DidMethod fromProto(BloockIdentityEntities.DidMethod method) {
    switch (method) {
      case POLYGON_ID:
        return DidMethod.PolygonID;
      case POLYGON_ID_TEST:
        return DidMethod.PolygonIDTest;
      default:
        return DidMethod.PolygonID;
    }
  }

  public BloockIdentityEntities.DidMethod toProto() {
    switch (this) {
      case PolygonID:
        return BloockIdentityEntities.DidMethod.POLYGON_ID;
      case PolygonIDTest:
        return BloockIdentityEntities.DidMethod.POLYGON_ID_TEST;
      default:
        return BloockIdentityEntities.DidMethod.POLYGON_ID;
    }
  }
}

File: bloock-core/src/identity/entity/did_method.rs
use crate::identity::IdentityError;

pub enum DidMethod {
    PolygonID,
    PolygonIDTest,
}

impl DidMethod {
    pub fn new(method: &str) -> Result<DidMethod, IdentityError> {
        match method {
            "polygon_id" => Ok(DidMethod::PolygonID),
            "polygon_id_test" => Ok(DidMethod::PolygonIDTest),
            _ => Err(IdentityError::InvalidDidMethod()),
        }
    }

    pub fn get_did_method(&self) -> String {
        match self {
            DidMethod::PolygonID => "polygon_id".to_string(),
            DidMethod::PolygonIDTest => "polygon_id_test".to_string(),
        }
    }
}
